torch/_subclasses/fake_tensor.py

```diff
@@ -1441,6 +1453,14 @@ def _validate_cache_key(
         # Avoid caching for any ops that would require a more sophisticated
         # caching implementation, e.g., data dependent ops or ops that modify
         # the inputs.
+        from torch._higher_order_ops.utils import registered_hop_fake_fns
+
+        if (
+            isinstance(func, torch._ops.HigherOrderOperator)
+            and func in registered_hop_fake_fns
```

Comment:
I'm not sure this is entirely correct, if the result of the HOP has data-dependent output shape or dynamic output shape then we need to bail out (https://github.com/pytorch/pytorch/blob/2fcfae72b4fc478dc7922b1efc637d9cb64dd42f/torch/_subclasses/fake_tensor.py#L1444-L1448)

Reply:
Now, I remember why I needed a `validator` function. Ah.

Reply:
Adding new logic that goes through the subgraph nodes and checks each of them.